### DISCUSSION TITLE
Editing a commit's metadata

### DISCUSSION THREAD

**Discussion by lenormf** [gitpython-developers/GitPython#1143] [Q&A] [answered]
Hi!

Is it possible to edit a commit's metadata (author, committer etc) using only a `Commmit` object?

I see that we can get a list of commit instances that reflect the history of a target branch, but could I change the attributes of those instances (e.g. the `author` attribute) and have the changes be applied to the repository (on the fly, or using some kind of `repo.commit()` function)?

Thanks!

**Comment by Byron** [comment, marked as answer]
The git commit history is immutable and thus append-only. Adjusting anything requires rewriting parts of the history.

This can be done easiest using `git rebase -i` or `git-filter-branch`. Equivalent functionality could be implemented in GitPython, but it's certainly very close to the 'metal'. 

Even though this issue is closed, please feel free to add further comments or remarks.
